tokio-udp/src/frame.rs

```diff
@@ -33,79 +36,98 @@ pub struct UdpFramed<C> {
     flushed: bool,
 }
 
-impl<C: Decoder> Stream for UdpFramed<C> {
-    type Item = (C::Item, SocketAddr);
-    type Error = C::Error;
+impl<C: Decoder + Unpin> Stream for UdpFramed<C> {
+    type Item = Result<(C::Item, SocketAddr), C::Error>;
+
+    fn poll_next(self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Option<Self::Item>> {
+        let pin = self.get_mut();
 
-    fn poll(&mut self) -> Poll<Option<(Self::Item)>, Self::Error> {
-        self.rd.reserve(INITIAL_RD_CAPACITY);
+        pin.rd.reserve(INITIAL_RD_CAPACITY);
 
         let (n, addr) = unsafe {
             // Read into the buffer without having to initialize the memory.
-            let (n, addr) = try_ready!(self.socket.poll_recv_from(self.rd.bytes_mut()));
-            self.rd.advance_mut(n);
+            let res = ready!(Pin::new(&mut pin.socket).poll_recv_from_priv(cx, pin.rd.bytes_mut()));
+            let (n, addr) = res.unwrap();
+            pin.rd.advance_mut(n);
             (n, addr)
         };
         trace!("received {} bytes, decoding", n);
-        let frame_res = self.codec.decode(&mut self.rd);
-        self.rd.clear();
+        let frame_res = pin.codec.decode(&mut pin.rd);
+        pin.rd.clear();
         let frame = frame_res?;
         let result = frame.map(|frame| (frame, addr)); // frame -> (frame, addr)
+        let result = result.unwrap();
```

Comment:
why is there an unwrap here?

Reply:
Ah, my mistake, I forgot to go back and map those errors. Fixing now.